File: app/src/main/java/ru/divizdev/coinrate/rates/LocaleUtils.java
package ru.divizdev.coinrate.rates;

import java.math.BigDecimal;
import java.math.RoundingMode;
import java.text.DecimalFormat;
import java.text.NumberFormat;
import java.util.Locale;

/**
 * Created by diviz on 10.02.2018.
 */

public class LocaleUtils {

    private static final Locale RUSSIAN_LOCALE = new Locale("RU");
    private static final Locale ENGLISH_LOCALE = Locale.ENGLISH;
    public static final String SYMBOL_PERCENT = "%";
    private static Locale _currentLocale;

    public static void setCurrentLocale(Locale locale) {
        _currentLocale = locale;
    }

    public static Locale getCurrentLocale() {
        if (_currentLocale != null) {
            return _currentLocale;
        }
        return RUSSIAN_LOCALE;
    }

    private static int getCurrentScale() {
        return 2;
    }

    public static String formatBigDecimal(BigDecimal decimal) {
        BigDecimal decimalCopy = decimal.setScale(getCurrentScale(), RoundingMode.HALF_DOWN);
        DecimalFormat df = (DecimalFormat) NumberFormat.getInstance(getCurrentLocale());
        df.setMinimumFractionDigits(getCurrentScale());
        df.setGroupingUsed(true);
        return df.format(decimalCopy);
    }

}

File: app/src/main/java/ru/divizdev/coinrate/utils/LocaleUtils.kt
package ru.divizdev.coinrate.utils

import java.math.BigDecimal
import java.math.RoundingMode
import java.text.DecimalFormat
import java.text.NumberFormat
import java.util.*

/**
 * Created by diviz on 10.02.2018.
 */
object LocaleUtils {
    private val RUSSIAN_LOCALE = Locale("RU")
    private val ENGLISH_LOCALE = Locale.ENGLISH
    const val SYMBOL_PERCENT = "%"
    private var _currentLocale: Locale? = null
    var currentLocale: Locale?
        get() = if (_currentLocale != null) {
            _currentLocale
        } else RUSSIAN_LOCALE
        set(locale) {
            _currentLocale = locale
        }
    private val currentScale: Int
        private get() = 2

    fun formatBigDecimal(decimal: BigDecimal?): String {
        if(decimal == null) return ""
        val decimalCopy = decimal.setScale(currentScale, RoundingMode.HALF_DOWN)
        val df = NumberFormat.getInstance(currentLocale) as DecimalFormat
        df.minimumFractionDigits = currentScale
        df.isGroupingUsed = true
        return df.format(decimalCopy)
    }
}